ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IGNITION__RVIZ__PLUGINS__POSEARRAYDISPLAY_HPP_
+#define IGNITION__RVIZ__PLUGINS__POSEARRAYDISPLAY_HPP_
+
+#include <ignition/rendering.hh>
+
+#include <geometry_msgs/msg/pose_array.hpp>
+
+#include <QColor>
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "ignition/rviz/plugins/message_display_base.hpp"
+
+namespace ignition
+{
+namespace rviz
+{
+namespace plugins
+{
+struct PoseArrayVisual
```

Comment:
Why is this part of a new struct? Can you define these variables and method inside the class ?

Reply:
The idea was to abstract to the PoseArray visuals. Going through the code again, it looks like I am not using the abstraction as properly. I can move it to the main class.